packages/playwright/src/runner/dispatcher.ts

```diff
@@ -339,8 +340,11 @@ class JobDispatcher {
       return;
     }
     step.duration = params.wallTime - step.startTime.getTime();
-    if (params.error)
+    if (params.error) {
       step.error = params.error;
+      if (params.error.matcherResult?.suggestedRebaseline)
+        addSuggestedRebaseline(step.location!, params.error.matcherResult.suggestedRebaseline);
```

Comment:
Why not just traverse the results in the end of the run?

Reply:
Does not seem to be a difference between event-based and on-result.